src/main/java/com/spotify/folsom/client/DefaultRawMemcacheClient.java

```diff
@@ -259,6 +270,10 @@ public void write(final ChannelHandlerContext ctx, final Object msg,
                       final ChannelPromise promise)
         throws Exception {
       Request<?> request = (Request<?>) msg;
+      if (request.isCancelled()) {
```

Comment:
Yeah, it'd probably be rare that requests would cancelled before they're written anyway.